### PR TITLE
implement sorting using enum SortType instead of constants

### PR DESCRIPTION
**Pull request by Koliras**
[DEMO LINK](https://Koliras.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,6 +1,36 @@
...
 import 'bulma/css/bulma.css';
 import './App.scss';
+import classNames from 'classnames';
+
+enum SortType {
+  Default = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
```
Comment (Orpheus29) [inline src/App.tsx:11]: You may want to destructure SortType at once as we used to

File: src/App.tsx [hunk 2]
```diff
@@ -16,47 +46,78 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
```
Comment (Orpheus29) [inline src/App.tsx:48]: Do we really need this FC here?

**Review by Orpheus29 — APPROVED** [review]
GJ!

File: src/App.tsx [hunk 3]
```diff
@@ -16,47 +46,78 @@ export const goodsFromServer = [
...
+  const [sortField, setSortField] = useState(SortType.Default);
```
Comment (roman-mirzoian) [inline src/App.tsx:49]:
```suggestion
  const [sortType, setSortType] = useState(SortType.Default);
```

File: src/App.tsx [hunk 4]
```diff
@@ -16,47 +46,78 @@ export const goodsFromServer = [
...
-        >
-          Reset
-        </button>
+        {isSorted && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setDefault();
+            }}
```
Comment (roman-mirzoian) [inline src/App.tsx:105]:
```suggestion
            onClick={setDefault}
```

File: src/App.tsx [hunk 5]
```diff
@@ -16,47 +46,78 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getOrderedGoods(
+    goodsFromServer, sortField, isReversed,
+  );
+  const isSorted = !!(sortField || isReversed);
+  const setDefault = () => {
```
Comment (roman-mirzoian) [inline src/App.tsx:55]: We need to choose a better name for this handler:
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

File: src/App.tsx [hunk 6]
```diff
@@ -1,6 +1,36 @@
...
 import './App.scss';
+import classNames from 'classnames';
+
+enum SortType {
+  Default = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+function getOrderedGoods(
```
Comment (roman-mirzoian) [inline src/App.tsx:12]:
```suggestion
function getSortedGoods(
```

File: src/App.tsx [hunk 7]
```diff
@@ -16,47 +46,78 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getOrderedGoods(
```
Comment (roman-mirzoian) [inline src/App.tsx:51]:
```suggestion
  const sortedGoods = getSortedGoods(
```

**Review by roman-mirzoian — CHANGES_REQUESTED** [review]
Good job, but there is quite a bit of work to be done.

File: src/App.tsx [hunk 8]
```diff
@@ -16,47 +46,76 @@ export const goodsFromServer = [
...
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames(
+            'button',
+            'is-warning',
+            { 'is-light': !isReversed },
+          )}
+          onClick={() => setIsReversed(!isReversed)}
```
Comment (Moroz-Dmytro) [inline src/App.tsx:94]:
```suggestion
          onClick={() => setIsReversed(isReversed => !isReversed)}
```

File: src/App.tsx [hunk 9]
```diff
@@ -16,47 +46,76 @@ export const goodsFromServer = [
...
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {sortedGoods.map(good => (
+          <li
+            data-cy="Good"
+            key={good}
+          >
```
Comment (Moroz-Dmytro) [inline src/App.tsx:115]:
```suggestion
          <li data-cy="Good" key={good}>
```

**Review by Moroz-Dmytro — APPROVED** [review]

**Pull request closed without merging**